### PR TITLE
maint: implementing autoapi and general maintenance

### PR DESCRIPTION
**Pull request by clatapie**

### CODE REVIEW

**Comment by clatapie** [comment]
The `ansys/pre-commit hook` doesn't seem to accept multiple files to be avoided. I tried using regex expression but didn't work either. Pinging @klmcadams for visibility.

**Comment by klmcadams** [comment]
@clatapie Thanks for letting me know! I’ll look into this

File: .coveragerc [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
```
Comment (clatapie) [inline .coveragerc:22]:
```suggestion
```

File: .gitattributes [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
```
Comment (clatapie) [inline .gitattributes:22]:
```suggestion
```

**Comment by clatapie** [comment]
Another PR will follow to add all the `ansys/pre-commit hook`

File: src/pyconverter/xml2py/__init__.py [hunk 3]
```diff
@@ -1,5 +1,3 @@
-# Copyright (c) 2023 ANSYS, Inc. All rights reserved.
-
 """
```
Comment (clatapie) [inline src/pyconverter/xml2py/__init__.py:1]:
```suggestion
# Copyright (c) 2024 ANSYS, Inc. All rights reserved.

"""
```

File: src/pyconverter/xml2py/ast_tree.py [hunk 4]
```diff
@@ -1,5 +1,3 @@
-# Copyright (c) 2023 ANSYS, Inc. All rights reserved.
-
 import logging
```
Comment (clatapie) [inline src/pyconverter/xml2py/ast_tree.py:1]:
```suggestion
# Copyright (c) 2024 ANSYS, Inc. All rights reserved.

import logging
```

File: src/pyconverter/xml2py/cli.py [hunk 5]
```diff
@@ -1,5 +1,3 @@
-# Copyright (c) 2023 ANSYS, Inc. All rights reserved.
-
 """Command Line Interface for PyConverter-XML2Py."""
```
Comment (clatapie) [inline src/pyconverter/xml2py/cli.py:1]:
```suggestion
# Copyright (c) 2024 ANSYS, Inc. All rights reserved.

"""Command Line Interface for PyConverter-XML2Py."""
```

File: src/pyconverter/xml2py/custom_functions.py [hunk 6]
```diff
@@ -1,5 +1,3 @@
-# Copyright (c) 2023 ANSYS, Inc. All rights reserved.
-
 import glob
```
Comment (clatapie) [inline src/pyconverter/xml2py/custom_functions.py:1]:
```suggestion
# Copyright (c) 2024 ANSYS, Inc. All rights reserved.

import glob
```

File: src/pyconverter/xml2py/directory_format.py [hunk 7]
```diff
@@ -1,5 +1,3 @@
-# Copyright (c) 2023 ANSYS, Inc. All rights reserved.
-
 import argparse
```
Comment (clatapie) [inline src/pyconverter/xml2py/directory_format.py:1]:
```suggestion
# Copyright (c) 2024 ANSYS, Inc. All rights reserved.

import argparse
```

File: src/pyconverter/xml2py/download.py [hunk 8]
```diff
@@ -1,6 +1,3 @@
-"""Functions to download template datasets from the pyconverter-xml2py repository.
-"""
-
 import os
```
Comment (clatapie) [inline src/pyconverter/xml2py/download.py:1]:
```suggestion
# Copyright (c) 2024 ANSYS, Inc. All rights reserved.

import os
```

File: src/pyconverter/xml2py/load_xml_doc.py [hunk 9]
```diff
@@ -1,5 +1,3 @@
-# Copyright (c) 2023 ANSYS, Inc. All rights reserved.
-
 import glob
```
Comment (clatapie) [inline src/pyconverter/xml2py/load_xml_doc.py:1]:
```suggestion
# Copyright (c) 2024 ANSYS, Inc. All rights reserved.

import glob
```

File: src/pyconverter/xml2py/version_variables.py [hunk 10]
```diff
@@ -1,6 +1,3 @@
-# Copyright (c) 2023 ANSYS, Inc. All rights reserved.
-
-
 class Autogenerateddirectory:
```
Comment (clatapie) [inline src/pyconverter/xml2py/version_variables.py:1]:
```suggestion
# Copyright (c) 2024 ANSYS, Inc. All rights reserved.

class Autogenerateddirectory:
```

File: src/pyconverter/xml2py/writer.py [hunk 11]
```diff
@@ -1,5 +1,3 @@
-# Copyright (c) 2023 ANSYS, Inc. All rights reserved.
-
 import glob
```
Comment (clatapie) [inline src/pyconverter/xml2py/writer.py:1]:
```suggestion
# Copyright (c) 2024 ANSYS, Inc. All rights reserved.

import glob
```

File: src/pyconverter/xml2py/download.py [hunk 12]
```diff
@@ -1,5 +1,4 @@
-"""Functions to download template datasets from the pyconverter-xml2py repository.
-"""
+# Copyright (c) 2024 ANSYS, Inc. All rights reserved.
```
Comment (clatapie) [inline src/pyconverter/xml2py/download.py:1]:
```suggestion
# Copyright (c) 2024 ANSYS, Inc. All rights reserved.

"""Functions to download template datasets from the pyconverter-xml2py repository.
"""
```

**Review by RobPasMue — APPROVED** [review]
LGTM!

**Pull request merged**